src/queue.ts

```diff
@@ -0,0 +1,1052 @@
+
+        return context.evalExprToJsonPathOrLiteral(
+          input,
+          (input, { addState }) => {
+            if (ASLGraph.isJsonPath(input)) {
+              addState({
+                Type: "Pass",
+                ResultPath: `${input.jsonPath}.QueueUrl`,
+                Result: this.resource.queueUrl,
+                Next: ASLGraph.DeferNext,
+              });
+              if (this.serializer.dataType === DataType.Json) {
+                // request.MessageBody.value = JSON.stringify(request.MessageBody)
+                addState({
+                  Type: "Pass",
+                  ResultPath: `${input.jsonPath}.MessageBody`,
+                  Parameters: {
+                    "value.$":
+                      "States.JsonToString(${input.jsonPath}.MessageBody)",
+                  },
+                  Next: ASLGraph.DeferNext,
+                });
+                // it's unfortunate that we have to do this
+                // request.MessageBody = request.MessageBody.value
+                addState({
+                  Type: "Pass",
+                  InputPath: `${input.jsonPath}.MessageBody.value`,
+                  OutputPath: `${input.jsonPath}.MessageBody`,
+                  Next: ASLGraph.DeferNext,
+                });
+              }
+              return context.stateWithHeapOutput({
+                Type: "Task",
+                Resource: "arn:aws:states:::aws-sdk:sqs:sendMessage",
+                Parameters: {
+                  "MessageBody.$": input.jsonPath,
+                  QueueUrl: this.queueUrl,
+                },
+                Next: ASLGraph.DeferNext,
+              });
+            } else if (
+              input.value !== null &&
+              typeof input.value === "object"
+            ) {
+              if (this.serializer.dataType === DataType.Json) {
+                // when the data type is Json, we need to serialize the MessageBody to JSON
+                let messageBodyJsonPath: string;
+                if ("MessageBody" in input.value) {
+                  messageBodyJsonPath = context.newHeapVariable();
+                  addState({
+                    Type: "Pass",
+                    Parameters: input.value.MessageBody,
+                    ResultPath: messageBodyJsonPath,
+                    Next: ASLGraph.DeferNext,
+                  });
+                } else if ("MessageBody.$" in input.value) {
+                  messageBodyJsonPath = input.value["MessageBody.$"];
+                } else {
+                  throw new SynthError(
+                    ErrorCodes.Invalid_Input,
+                    `the property 'MessageBody' is required in SendMessageRequest`
+                  );
+                }
+                return context.stateWithHeapOutput({
+                  Type: "Task",
+                  Resource: "arn:aws:states:::aws-sdk:sqs:sendMessage",
+                  Parameters: {
+                    ...input.value,
```

Comment:
if Message is a constant, you'll end up with both `Message` and `Message.$`, which will fail.